src/reducers/teacherInfo.js

```diff
@@ -0,0 +1,14 @@
+export default function(state={}, action) {
+  switch (action.type) {
+    case 'SET_TEACHER_INFO':
+      return isValidPayload(action)
+        ? action.payload.teacherInfo
+        : state;
+  }
+
+  return state;
+}
+
+function isValidPayload(action) {
+  return action.payload.teacherInfo !== null;
```

Comment:
null og undefined er ikke det samme. Problemet er at når den er undefined, så blir isValidPayload true.

Prøv å bytte ut !== med != (et av de få tilfellene hvor det er greit å bare bruke != i javascript), for da får den med både null og undefined.